src/app/replayer/replayer.ml

```diff
@@ -4,21 +4,38 @@ open Core
 open Async
 open Coda_base
 
-(* given a target block B to replay to:
+(* identify a target block B containing staking and next epoch ledgers
+   to be used in a hard fork, by giving its state hash
+
+   from B, we choose a predecessor block B_fork, which is the block to
+   fork from
+
+   we replay all commands, one by one, from the genesis block through
+   B_fork
+
+   when the Merkle root of the replay ledger matches one of the
+   epoch ledger hashes, we make a copy of the replay ledger to
+   become that target epoch ledger
+
+   when all commands from a block have been replayed, we verify
+   that the Merkle root of the replay ledger matches the stored
+   ledger hash in the archive database
 
-   target state hash  = protocol state hash in B
-   target proof       = blockchain SNARK proof of the protocol state in B
 *)
+
 type input =
-  { target_state_hash: State_hash.t
-  ; target_proof: Proof.t
+  { target_epoch_ledgers_state_hash: State_hash.t
   ; genesis_ledger: Runtime_config.Ledger.t }
 [@@deriving yojson]
 
 type output =
-  { target_state_hash: State_hash.t
-  ; target_proof: Proof.t
-  ; target_ledger: Runtime_config.Ledger.t }
+  { target_epoch_ledgers_state_hash: State_hash.t
+  ; target_fork_state_hash: State_hash.t
+  ; target_genesis_ledger: Runtime_config.Ledger.t
+  ; target_staking_epoch_ledger: Runtime_config.Ledger.t
+  ; target_staking_seed: Epoch_seed.t
+  ; target_next_epoch_ledger: Runtime_config.Ledger.t
+  ; target_next_seed: Epoch_seed.t }
```

Comment:
Combine the epoch data to this newly added  type `Runtime_config.Epoch_data.t`?

Reply:
OK, I'm copying that type from #6632 to this PR.